server/src/daemon/collection.rs

```diff
@@ -0,0 +1,217 @@
+        };
+
+        let item = self
+            .keyring
+            .create_item(label, &attributes, secret.value(), replace)
+            .await
+            .map_err::<ServiceError, _>(From::from)?;
+        *self.item_counter.write().await += 1;
+
+        let item = super::item::Item::new(
+            item,
+            parameters,
+            content_type,
+            *self.item_counter.read().await,
+            self.path(),
+            Arc::clone(&self.keyring),
+            Arc::clone(&self.manager),
+        )
+        .await;
+        let path = OwnedObjectPath::from(item.path());
+        self.items.write().await.push(item.clone());
+        object_server.at(&path, item).await.unwrap();
+
+        // perform prompt
+        self.manager.lock().unwrap().update_prompts_counter();
+        let prompt = Prompt::new(
+            Arc::clone(&self.manager),
+            self.manager.lock().unwrap().prompts_counter(),
+            None,
+        );
+        object_server
+            .at(prompt.path().to_owned(), prompt.to_owned())
+            .await?;
+
+        // signal
+        Self::item_created(&ctxt, path.as_ref()).await?;
+
+        Ok((path, prompt.path().to_owned()))
+    }
+
+    #[zbus(property, name = "Items")]
+    pub async fn items(&self) -> Vec<OwnedObjectPath> {
+        self.items
+            .read()
+            .await
+            .iter()
+            .map(|item| OwnedObjectPath::from(item.path()))
+            .collect()
+    }
+
+    #[zbus(property, name = "Label")]
+    pub fn label(&self) -> &str {
+        &self.label
+    }
+
+    #[zbus(property, name = "Locked")]
+    pub fn locked(&self) -> bool {
+        self.locked.load(std::sync::atomic::Ordering::Relaxed)
+    }
+
+    #[zbus(property, name = "Created")]
+    pub fn created_as_secs(&self) -> u64 {
+        self.created.as_secs()
+    }
+
+    #[zbus(property, name = "Modified")]
+    pub fn modified_as_secs(&self) -> u64 {
+        self.modified.as_secs()
+    }
+
+    #[zbus(signal)]
+    pub async fn item_created(ctxt: &SignalContext<'_>, item: ObjectPath<'_>) -> zbus::Result<()>;
+
+    #[zbus(signal)]
+    pub async fn item_deleted(ctxt: &SignalContext<'_>, item: ObjectPath<'_>) -> zbus::Result<()>;
+
+    #[zbus(signal)]
+    pub async fn item_changed(ctxt: &SignalContext<'_>, item: ObjectPath<'_>) -> zbus::Result<()>;
+}
+
+impl Collection {
+    pub fn new(
+        label: &str,
+        alias: &str,
+        created: Duration,
+        keyring: Arc<Keyring>,
+        manager: Arc<Mutex<ServiceManager>>,
+    ) -> Self {
+        Self {
+            items: Default::default(),
+            label: label.to_owned(),
+            alias: Arc::new(RwLock::new(alias.to_owned())),
+            locked: Arc::new(AtomicBool::new(false)),
+            modified: created,
+            item_counter: Arc::new(RwLock::new(0)),
+            path: OwnedObjectPath::try_from(format!("{}{}", SECRET_COLLECTION_PREFIX, label))
+                .unwrap(),
```

Comment:
Same, those two should be handled when creating the collection

Reply:
Moved incrementing the counter to Prompt constructors.